src/plugins/data/public/search/aggs/metrics/percentiles_fn.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { Assign } from '@kbn/utility-types';
+import { ExpressionFunctionDefinition } from '../../../../../expressions/public';
+import { AggExpressionType, AggExpressionFunctionArgs, METRIC_TYPES } from '../';
+import { getParsedValue } from '../utils/get_parsed_value';
+
+const fnName = 'aggPercentiles';
+
+type Input = any;
+type AggArgs = AggExpressionFunctionArgs<typeof METRIC_TYPES.PERCENTILES>;
+type Arguments = Assign<AggArgs, { percents?: number[] }>;
+type Output = AggExpressionType;
+type FunctionDefinition = ExpressionFunctionDefinition<typeof fnName, Input, Arguments, Output>;
+
+export const aggPercentiles = (): FunctionDefinition => ({
+  name: fnName,
+  help: i18n.translate('data.search.aggs.function.metrics.percentiles.help', {
+    defaultMessage: 'Generates a serialized agg config for a percentiles agg',
+  }),
+  type: 'agg_type',
+  args: {
+    id: {
+      types: ['string'],
+      help: i18n.translate('data.search.aggs.metrics.percentiles.id.help', {
+        defaultMessage: 'ID for this aggregation',
+      }),
+    },
+    enabled: {
+      types: ['boolean'],
+      default: true,
+      help: i18n.translate('data.search.aggs.metrics.percentiles.enabled.help', {
+        defaultMessage: 'Specifies whether this aggregation should be enabled',
+      }),
+    },
+    schema: {
+      types: ['string'],
+      help: i18n.translate('data.search.aggs.metrics.percentiles.schema.help', {
+        defaultMessage: 'Schema to use for this aggregation',
+      }),
+    },
+    field: {
+      types: ['string'],
+      required: true,
+      help: i18n.translate('data.search.aggs.metrics.percentiles.field.help', {
+        defaultMessage: 'Field to use for this aggregation',
+      }),
+    },
+    percents: {
+      types: ['number'],
+      multi: true,
+      default: [1, 5, 25, 50, 75, 95, 99],
```

Comment:
@lukeelmers I don't see any reasons to duplicate the default params in our expressions functions.  It's enough to mark it as optional field.

Reply:
@alexwizp Yes, I was thinking about this and I agree we can leave the defaults out for the time being as well, for a few reasons:

1. It feels like the duplication could lead to easy-to-miss bugs in the future (if one default changes and we forget to update the other).
2. Most folks are not going to be constructing an expression relying on these functions from scratch -- the primary use case is taking an existing `AggConfig` and calling `toExpressionAst()` on it, which is going to generate the expression AST with the defaults already filled in.
3. It's easy to go back and add them later if we change our minds.

Reply:
Removed defaults from functions.